lib/glific/erp.ex

```diff
@@ -113,33 +114,28 @@ defmodule Glific.ERP do
 
   @spec create_or_update_address(map(), String.t()) :: :ok | {:error, String.t()}
   defp create_or_update_address(registration, customer_name) do
-    billing_exists? = address_exists?(customer_name, "Billing")
-    permanent_exists? = address_exists?(customer_name, "Permanent/Registered")
-
     current_address = registration.org_details["current_address"]
     registered_address = registration.org_details["registered_address"]
     are_addresses_same = compare_addresses(current_address, registered_address)
 
-    case (if billing_exists? do
-            update_address(current_address, "Billing", customer_name)
-          else
-            create_address(current_address, "Billing", customer_name)
-          end) do
-      {:ok, _} -> :ok
+    with {:ok, _} <- handle_address(current_address, "Billing", customer_name),
+         {:ok, _} <-
+           if(are_addresses_same,
+             do: {:ok, nil},
+             else: handle_address(registered_address, "Permanent/Registered", customer_name)
+           ) do
+      :ok
+    else
       {:error, reason} -> {:error, reason}
     end
+  end
 
-    if are_addresses_same do
-      :ok
+  @spec handle_address(map(), String.t(), String.t()) :: {:ok, map()} | {:error, String.t()}
+  defp handle_address(address, type, customer_name) do
```

Comment:
can we try to have a function name for `handle_address` which provides more clarity on what it does, something like `do_create_or_update_address` for ex.